recipe-server/client/control_new/components/recipes/CloneRecipePage.js

```diff
@@ -0,0 +1,117 @@
+import { message } from 'antd';
+import autobind from 'autobind-decorator';
+import { Map } from 'immutable';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link, push as pushAction } from 'redux-little-router';
+
+import { SimpleLoadingOverlay } from 'control_new/components/common/LoadingOverlay';
+import RecipeForm from 'control_new/components/recipes/RecipeForm';
+import QueryRecipe from 'control_new/components/data/QueryRecipe';
+
+import { createRecipe as createAction } from 'control_new/state/recipes/actions';
+import { getRecipeFromURL } from 'control_new/state/recipes/selectors';
+
+@connect(
+  state => {
+    const {
+      isLatestRevision,
+      recipe,
+      recipeId,
+      revisionId,
+    } = getRecipeFromURL(state);
+
+    return {
+      isLatestRevision,
+      recipe,
+      recipeId,
+      revisionId,
+    };
+  },
+  {
+    push: pushAction,
+    createRecipe: createAction,
+  },
+)
+@autobind
+export default class CloneRecipePage extends React.Component {
+  static propTypes = {
+    createRecipe: PropTypes.func.isRequired,
+    isLatestRevision: PropTypes.bool,
+    recipeId: PropTypes.number.isRequired,
+    recipe: PropTypes.instanceOf(Map),
+    revisionId: PropTypes.string.isRequired,
+  };
+
+  static defaultProps = {
+    recipe: null,
+    isLatestRevision: false,
+  };
+
+  state = {
+    formErrors: undefined,
+  };
+
+  /**
+   * Update the existing recipe and display a message.
+   */
+  async handleSubmit(values) {
+    const { push } = this.props;
+
+    try {
+      const newId = await this.props.createRecipe(values);
+
+      message.success('Recipe saved');
+      this.setState({
+        formErrors: undefined,
+      });
+
+      push(`/recipe/${newId}`);
+    } catch (error) {
+      message.error(
+        'Recipe cannot be saved. Please correct any errors listed in the form below.',
+      );
+
+      this.setState({
+        formErrors: error.data || error,
+      });
+    }
+  }
+
+  render() {
+    const { recipe, recipeId, isLatestRevision, revisionId } = this.props;
+    const recipeName = recipe.get('name');
+
+    // Remove the 'name' field value.
+    const displayedRecipe = recipe.set('name');
+
+    // URL and text for the "Cloning this from [a recipe]" link
+    const originalRecipeURL = `/recipe/${recipeId}${isLatestRevision ? '' : `/rev/${revisionId}`}`;
+
+    // Only display revision hash if we're _not_ on the latest version.
+    const revisionAddendum = isLatestRevision ? '' : `(revision ${revisionId.slice(0, 7)}...)`;
+    const originalRecipeText = `"${recipeName}" ${revisionAddendum}`;
+
+    return (
+      <div className="clone-page">
+        <QueryRecipe pk={recipeId} />
+        <SimpleLoadingOverlay isVisible={!recipeName}>
+          <h2>New Recipe</h2>
```

Comment:
I think it's better to signify that it's a `new` recipe versus a `cloned` recipe. 'Cloning' here is essentially using the values from one recipe to autofill the New Recipe form. Users can then mess with the values, the original history is not brought over, etc. There's no connection to the original recipe and the new recipe beyond any values that were left untouched.

To the end user, the only difference between hitting `new recipe` and `clone` is that on the clone page it has `Cloning from "some recipe"` and the fields are pre-populated. At the end of the day, you're still just creating a new recipe.